fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/services/ContainerServiceImpl.java

```diff
@@ -61,10 +63,22 @@ public Container findOrCreate(final Session session, final String path) {
 
             if (node.isNew()) {
                 initializeNewObjectProperties(node);
+
+                final Node parent = node.getDepth() > 0 ? node.getParent() : null;
+
+                if (parent != null) {
+                    touch(parent);
+                    touchLdpMembershipResource(node);
+                }
+            }
+
+            if (node.isNew()) {
+                touch(node);
             }
 
             return new ContainerImpl(node);
         } catch (final RepositoryException e) {
+            LOGGER.info("Error: {}", e);
```

Comment:
I think that "debug mode" actually _is_ `DEBUG` level logging. Here's my proposed rule: log informative messages that enable immediate concrete action to correct the specific problem, intended for integrators, administrators, and programmers, at `ERROR` level. Log stacktraces and other diagnostic information that would be primarily of interest to integrators and programmers, for the purpose of understanding how to correct the integration or code to avoid the problem in future, at `DEBUG` level. Log deep reporting of control flow and logic that would be primarily of interest to programmers at `TRACE` level.